File: 153.Find-Minimum-in-Rotated-Sorted-Array/dnc.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        mid = len(nums) // 2
        l = self.findMin(nums[:mid])
        r = self.findMin(nums[mid:])
        return min(l, r)

File: 153.Find-Minimum-in-Rotated-Sorted-Array/dnc.rs
impl Solution {
    fn dnc(nums: &[i32]) -> i32 {
        if let [only] = nums {
            return *only;
        }

        let mid = nums.len() / 2;
        let l = Self::dnc(&nums[..mid]);
        let r = Self::dnc(&nums[mid..]);
        l.min(r)
    }

    pub fn find_min(nums: Vec<i32>) -> i32 {
        Self::dnc(&nums)
    }
}
